airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/split_secrets/SecretsHelpersTest.java

```diff
@@ -0,0 +1,219 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.config.persistence.split_secrets;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import io.airbyte.config.persistence.split_secrets.test_cases.ArrayTestCase;
+import io.airbyte.config.persistence.split_secrets.test_cases.NestedObjectTestCase;
+import io.airbyte.config.persistence.split_secrets.test_cases.OneOfTestCase;
+import io.airbyte.config.persistence.split_secrets.test_cases.OptionalPasswordTestCase;
+import io.airbyte.config.persistence.split_secrets.test_cases.SimpleTestCase;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class SecretsHelpersTest {
+
+  public static final UUID WORKSPACE_ID = UUID.fromString("e0eb0554-ffe0-4e9c-9dc0-ed7f52023eb2");
+
+  // use a fixed sequence of UUIDs so it's easier to have static files for the test cases
+  public static final List<UUID> UUIDS = List.of(
+      UUID.fromString("9eba44d8-51e7-48f1-bde2-619af0e42c22"),
+      UUID.fromString("2c2ef2b3-259a-4e73-96d1-f56dacee2e5e"),
+      UUID.fromString("1206db5b-b968-4df1-9a76-f3fcdae7e307"),
+      UUID.fromString("c03ef566-79a7-4e77-b6f3-d23d2528f25a"),
+      UUID.fromString("35f08b15-bfd9-44fe-a8c7-5aa9e156c0f5"),
+      UUID.fromString("159c0b6f-f9ae-48b4-b7f3-bcac4ba15743"),
+      UUID.fromString("71af9b74-4e61-4cff-830e-3bf1ec18fbc0"),
+      UUID.fromString("067a62fc-d007-44dd-a8f6-0fd10823713d"),
+      UUID.fromString("c4967ac9-0856-4733-a21e-1d51ca8f254d"));
+
+  private static Stream<Arguments> provideTestCases() {
+    return Stream.of(
+        new OptionalPasswordTestCase(),
+        new SimpleTestCase(),
+        new NestedObjectTestCase(),
+        new OneOfTestCase(),
+        new ArrayTestCase()
+    // new ArrayOneOfTestCase() todo: support this test case
+    ).map(Arguments::of);
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideTestCases")
+  void testSplit(SecretsTestCase testCase) {
+    final var uuidIterator = UUIDS.iterator();
+    final var inputConfig = testCase.getFullConfig();
+    final var inputConfigCopy = inputConfig.deepCopy();
+    final var splitConfig = SecretsHelpers.split(
```

Comment:
Now that I'm seeing the split method outside of the class it's defined in, I am realizing the name is a bit unclear.  Naming the methods splitConfig and combineConfig might be more obvious.  That way the meaning is apparent even if it's inlined without a matching variable name.